__tests__/commands/install.js

```diff
@@ -71,6 +71,18 @@ test.concurrent("removes extraneous files that aren't in module or artifacts", a
   }, create);
 });
 
+test.concurrent('--production flag ignores dev dependencies', () => {
```

Comment:
we also need a test for https://github.com/yarnpkg/yarn/issues/2073 - when a dev dependency is a transitive dependency of a prod dependency.